pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -684,10 +684,21 @@ func (gc *GarbageCollector) processItem(item *node) error {
 }
 
 func (gc *GarbageCollector) Run(workers int, stopCh <-chan struct{}) {
+	glog.Infof("The garbage collector is starting to run")
 	for _, monitor := range gc.monitors {
 		go monitor.controller.Run(stopCh)
 	}
 
+	wait.PollInfinite(10*time.Second, func() (bool, error) {
+		for _, monitor := range gc.monitors {
+			if !monitor.controller.HasSynced() {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	glog.Infof("All resource monitors of the garbage collector have been synced")
```

Comment:
"Garbage Collector: All monitored resources synced. Proceeding to collect garbage."